pkg/entities/record.go

```diff
@@ -98,17 +99,25 @@ func (b *baseRecord) GetFieldCount() uint16 {
 	return b.fieldCount
 }
 
-func (b *baseRecord) GetOrderedElementList() []*InfoElementWithValue {
+func (b *baseRecord) GetOrderedElementList() []InfoElementWithValue {
 	return b.orderedElementList
 }
 
-func (b *baseRecord) GetInfoElementWithValue(name string) (*InfoElementWithValue, bool) {
-	for _, element := range b.orderedElementList {
+func (b *baseRecord) GetInfoElementWithValue(name string) (*InfoElementWithValue, int, bool) {
+	for i, element := range b.orderedElementList {
 		if element.Element.Name == name {
-			return element, true
+			return &element, i, true
 		}
 	}
-	return nil, false
+	return nil, 0, false
+}
+
+func (b *baseRecord) SetInfoElementWithValue(index int, infoElement *InfoElementWithValue) error {
```

Comment:
I'm wondering why we use pointer instead of value here since we will pass value to element list. Does it have something to do with memory consumption?

Reply:
Yes I tried that and saw more memory consumption when passing the value. 
Let me try couple of things here and check if there is any change.

Reply:
Previously when I tried, I moved to value in `GetInfoElementWithValue`.. that increased memory consumption. When I did it only with `SetInfoElementWithValue`, there is slight improvement in the heap usage. 
Moving it to value here.